validate-pr/validate_pr.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+# bash strict mode -- http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+if ! command -v hub; then
+    echo 'Error: GitHub command line tool is not installed.' >&2
+fi
+
+PR_NUMBER=$1
+HEAD_BRANCH=$2 # fb_featureName
+BASE_BRANCH=$3 # develop
+PR_TITLE=$4
+ERROR_FILE="errors.tmp"
+
+if [ -z "$BASE_BRANCH" ] || [ -z "$HEAD_BRANCH" ] || [ -z "$PR_NUMBER" ] || [ -z "$PR_TITLE" ]; then
+    echo "PR info not specified" >&2
+    exit 1
+fi
+
+if [ -f "$ERROR_FILE" ]; then
+    rm "$ERROR_FILE"
+fi
+
+REL_PATTERN='^release([0-9]+\.[0-9]+)$'           # release20.11
+SNAP_PATTERN='^release([0-9]+\.[0-9]+)-SNAPSHOT$' # release20.11-SNAPSHOT
+FF_PATTERN='^ff_([0-9]+\.[0-9]+).+'               # ff_20.11.0
+FB_PATTERN='^fb_.+'                               # fb_newFeature_12345
+RFB_PATTERN='^([0-9]+\.[0-9]+)_fb_.+'             # 20.11_fb_backportFeature_12345
+
+{
+    # 'fb_*' should target 'develop'
+    if [[ "$HEAD_BRANCH" =~ $FB_PATTERN ]]; then
+        if [ "$BASE_BRANCH" != "develop" ]; then
+            echo "Expect PR from \`${HEAD_BRANCH}\` to target \`develop\`, not \`${BASE_BRANCH}\`"
```

Comment:
If I ever have to do anything more than twice, I usually turn that into a function. So all the instances of this error message could look like: `unexpected_pr_target "${HEAD_BRANCH}" 'develop' "${BASE_BRANCH}"`